File: backend/src/main/java/com/back/back9/domain/wallet/dto/WalletBalanceResponse.java
package com.back.back9.domain.wallet.dto;

import com.back.back9.domain.wallet.entity.Wallet;

import java.math.BigDecimal;
// 지갑 잔액 응답 DTO
public record  WalletBalanceResponse (
        int walletId,
        int userId,
        String address,
        BigDecimal balance
) {
    public WalletBalanceResponse(int walletId, int userId, String address, BigDecimal balance) {
        this.walletId = walletId;
        this.userId = userId;
        this.address = address;
        this.balance = balance;
    }

    // 정적 팩토리 메서드
    public WalletBalanceResponse of(int walletId, int userId, String address, BigDecimal balance) {
        return new WalletBalanceResponse(walletId, userId, address, balance);
    }
    // Wallet 엔티티를 WalletBalanceResponse로 변환하는 정적 팩토리 메서드
    public static WalletBalanceResponse from(Wallet wallet){
        return new WalletBalanceResponse(
                wallet.getId(),
                wallet.getUserId(),
                wallet.getAddress(),
                wallet.getBalance()
        );
    }

}

File: backend/src/main/kotlin/com/back/back9/domain/wallet/dto/WalletBalanceResponse.kt
package com.back.back9.domain.wallet.dto

import com.back.back9.domain.wallet.entity.Wallet

data class WalletBalanceResponse(
    val walletId: Long,
    val userId: Long,
    val address: String,
    val balance: java.math.BigDecimal
) {
    companion object {
        fun of(walletId: Long, userId: Long, address: String, balance: java.math.BigDecimal) =
            WalletBalanceResponse(walletId, userId, address, balance)

        fun from(wallet: Wallet) = WalletBalanceResponse(
            walletId = wallet.id,
            userId = wallet.user.id,
            address = wallet.address,
            balance = wallet.balance.toBigDecimal()
        )
    }
}
